2022-10-11 03:20:28,603 INFO    Thread-8  :15967 [wandb_setup.py:_flush():75] Loading settings from /root/.config/wandb/settings
2022-10-11 03:20:28,604 INFO    Thread-8  :15967 [wandb_setup.py:_flush():75] Loading settings from wandb/settings
2022-10-11 03:20:28,604 INFO    Thread-8  :15967 [wandb_setup.py:_flush():75] Loading settings from environment variables: {'project': 'meme_experiments', 'entity': 'vietnguyen', 'root_dir': '/content/SSLMemes', 'sweep_id': '2axv9u8x', 'run_id': 'n5f6ey1p', 'sweep_param_path': '/content/SSLMemes/wandb/sweep-2axv9u8x/config-n5f6ey1p.yaml'}
2022-10-11 03:20:28,604 INFO    Thread-8  :15967 [wandb_setup.py:_flush():75] Inferring run settings from compute environment: {'program_relpath': 'src/train.py', 'program': '/content/SSLMemes/src/train.py'}
2022-10-11 03:20:28,604 INFO    Thread-8  :15967 [wandb_init.py:_log_setup():405] Logging user logs to /content/SSLMemes/wandb/run-20221011_032028-n5f6ey1p/logs/debug.log
2022-10-11 03:20:28,604 INFO    Thread-8  :15967 [wandb_init.py:_log_setup():406] Logging internal logs to /content/SSLMemes/wandb/run-20221011_032028-n5f6ey1p/logs/debug-internal.log
2022-10-11 03:20:28,604 INFO    Thread-8  :15967 [wandb_init.py:init():439] calling init triggers
2022-10-11 03:20:28,604 INFO    Thread-8  :15967 [wandb_init.py:init():443] wandb.init called with sweep_config: {'batch_size': 80, 'dropout': 0, 'epochs': 1, 'hidden_size': 256, 'learning_rate': 0.09230298472420188, 'optimizer': 'adam'}
config: {}
2022-10-11 03:20:28,605 INFO    Thread-8  :15967 [wandb_init.py:init():492] starting backend
2022-10-11 03:20:28,605 INFO    Thread-8  :15967 [backend.py:_multiprocessing_setup():101] multiprocessing start_methods=fork,spawn,forkserver, using: spawn
2022-10-11 03:20:28,629 INFO    Thread-8  :15967 [backend.py:ensure_launched():219] starting backend process...
2022-10-11 03:20:28,646 INFO    Thread-8  :15967 [backend.py:ensure_launched():225] started backend process with pid: 16052
2022-10-11 03:20:28,650 INFO    Thread-8  :15967 [wandb_init.py:init():501] backend started and connected
2022-10-11 03:20:28,669 INFO    Thread-8  :15967 [wandb_run.py:_config_callback():992] config_cb None None {'batch_size': 80, 'dropout': 0, 'epochs': 1, 'hidden_size': 256, 'learning_rate': 0.09230298472420188, 'optimizer': 'adam'}
2022-10-11 03:20:28,673 INFO    Thread-8  :15967 [wandb_init.py:init():565] updated telemetry
2022-10-11 03:20:28,714 INFO    Thread-8  :15967 [wandb_init.py:init():596] communicating run to backend with 30 second timeout
2022-10-11 03:20:35,783 INFO    Thread-8  :15967 [wandb_run.py:_on_init():1759] communicating current version
2022-10-11 03:20:35,860 INFO    Thread-8  :15967 [wandb_run.py:_on_init():1763] got version response upgrade_message: "wandb version 0.13.4 is available!  To upgrade, please run:\n $ pip install wandb --upgrade"

2022-10-11 03:20:35,860 INFO    Thread-8  :15967 [wandb_init.py:init():625] starting run threads in backend
2022-10-11 03:20:40,492 INFO    Thread-8  :15967 [wandb_run.py:_console_start():1733] atexit reg
2022-10-11 03:20:40,492 INFO    Thread-8  :15967 [wandb_run.py:_redirect():1606] redirect: SettingsConsole.REDIRECT
2022-10-11 03:20:40,494 INFO    Thread-8  :15967 [wandb_run.py:_redirect():1611] Redirecting console.
2022-10-11 03:20:40,495 ERROR   Thread-8  :15967 [wandb_run.py:_redirect():1670] Failed to redirect.
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/wandb_run.py", line 1663, in _redirect
    out_redir.install()
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/lib/redirect.py", line 709, in install
    _WSCH.add_fd(self._pipe_read_fd)
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/lib/redirect.py", line 659, in add_fd
    self._register()
  File "/usr/local/lib/python3.7/dist-packages/wandb/sdk/lib/redirect.py", line 644, in _register
    old_handler = signal.signal(signal.SIGWINCH, lambda *_: None)
  File "/usr/lib/python3.7/signal.py", line 47, in signal
    handler = _signal.signal(_enum_to_int(signalnum), _enum_to_int(handler))
ValueError: signal only works in main thread
2022-10-11 03:20:40,498 INFO    Thread-8  :15967 [wandb_init.py:init():664] run started, returning control to user process
2022-10-11 03:20:40,499 INFO    Thread-8  :15967 [wandb_run.py:_config_callback():992] config_cb None None {'img_feature_path': 'data/features/visualbert/', 'train_csv_path': 'data/splits/random/memotion_train.csv', 'val_csv_path': 'data/splits/random/memotion_val.csv', 'model_type': 'visualbert', 'model_path': 'uclanlp/visualbert-nlvr2-coco-pre', 'epoch': 100, 'eval_step': 100, 'amp': True, 'output_dir': './tmp', 'checkpoint_step': 100, 'random_seed': 42, 'resume_training': False, 'semi_supervised': False}
2022-10-11 03:22:11,776 INFO    Thread-8  :15967 [wandb_run.py:_config_callback():992] config_cb None None {}
